src/python/pants/backend/go/module.py

```diff
@@ -44,10 +45,22 @@ class ModuleDescriptor:
 
 @dataclass(frozen=True)
 class ResolvedGoModule:
+    # Address of the resolved go_module target.
     address: Address
+
+    # The go_module target.
+    target: Target
```

Comment:
The `address` can be computed by the `target` as `target.address`. Imo, cleaner to not duplicate the data like this. If you find you're using `target.address` a lot at call-sites, you could add an `@property def address()`.